edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProcessor.cs

```diff
@@ -59,7 +59,12 @@ public async Task<IReadOnlyList<ModuleLogMessage>> GetMessages(string id, Stream
             IRunnableGraph<Task<IImmutableList<ModuleLogMessage>>> graph = graphBuilder.GetMaterializingGraph(m => (ModuleLogMessage)m);
 
             IImmutableList<ModuleLogMessage> result = await graph.Run(this.materializer);
-            return result;
+            return filter.Tail.Match<IReadOnlyList<ModuleLogMessage>>(
+                t =>
+                {
+                    return result.Skip(Math.Max(0, result.Count - t)).ToList().AsReadOnly();
```

Comment:
Unfortunately, I don't think `Math.Abs()` will work here. More particularly in case `t > result.Count`, Abs() would return the difference.. that's not what I want.